GeneratorInterface/Hydjet2Interface/src/Hydjet2Hadronizer.cc

```diff
@@ -40,32 +40,40 @@ using namespace edm;
 using namespace std;
 using namespace gen;
 
-int Hydjet2Hadronizer::convertStatusForComponents(int sta, int typ) {
-  if (sta == 1 && typ == 0)
-    return 6;
-  if (sta == 1 && typ == 1)
-    return 7;
-  if (sta == 2 && typ == 0)
-    return 16;
-  if (sta == 2 && typ == 1)
-    return 17;
-
+int Hydjet2Hadronizer::convertStatusForComponents(int sta, int typ, int pySta) {
+  int st = -1;
+  if (typ == 0)  //soft
+    st = 2 - sta;
+  else if (typ == 1)
+    st = convertStatus(pySta);
+
```

Comment:
You can throw already here, instead that later at L63:
```C++
  if (st == -1)
     throw cms::Exception("ConvertStatus") << "Wrong status code!" << endl;
```

Reply:
Ok, thanks!